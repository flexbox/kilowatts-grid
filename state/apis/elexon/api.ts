import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import * as pn from "./pn";
import * as mels from "./mels";
import * as boalf from "./boalf";

export const BASE_API = `https://data.elexon.co.uk/bmrs/api/v1`;

export const elexonApi = createApi({
  reducerPath: "elexonApi",
  baseQuery: fetchBaseQuery({ baseUrl: BASE_API }),
  endpoints: (builder) => ({
    pn: builder.query({
      query: pn.query,
      transformResponse: pn.transformResponse
    }),
    mels: builder.query({
      query: mels.query,
      transformResponse: mels.transformResponse
    }),
    boalf: builder.query({
      query: boalf.query,
      transformResponse: boalf.transformResponse
    }),
  }),
});


export const { usePnQuery, useMelsQuery, useLazyMelsQuery, useBoalfQuery } = elexonApi;