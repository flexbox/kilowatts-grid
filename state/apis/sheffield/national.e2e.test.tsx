import { createStore } from "../../reducer";
import { sheffieldApi } from "./api";
import { renderHook, cleanup, waitFor } from "@testing-library/react-native";
import { Provider } from "react-redux";
import React from "react";

const { useNationalQuery } = sheffieldApi;

const wrapper = (props: any) => {
  return <Provider store={createStore()} children={props.children} />;
};

describe("state/apis/sheffield/national.e2e.test.ts", () => {
  test("useNationalQuery", async () => {
    const p = {
      start: "2023-12-01T12:00:00.000Z",
      end: "2023-12-01T12:30:00.000Z",
    };
    const r = renderHook(() => useNationalQuery(p), { wrapper });
    await waitFor(() => {
      expect(r.result.current.data).toBeDefined();
      expect(r.result.current.data?.length).toBe(2);
      expect(r.result.current.data).toMatchSnapshot();
    });

    r.unmount();
  });

  afterEach(async () => {
    await cleanup();
  });
});
