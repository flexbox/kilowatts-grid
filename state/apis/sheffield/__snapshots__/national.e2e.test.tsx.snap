// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state/apis/elexon/boalf.e2e.test.ts useBoalfQuery 1`] = `
[
  {
    "capacity": 15954.059,
    "output": 4186.84,
    "start": "2023-12-01T12:00:00Z",
  },
  {
    "capacity": 15954.059,
    "output": 4189.23,
    "start": "2023-12-01T12:30:00Z",
  },
]
`;

exports[`state/apis/sheffield/national.e2e.test.ts useNationalQuery 1`] = `
[
  {
    "capacity": 15954.059,
    "output": 4186.84,
    "start": "2023-12-01T12:00:00Z",
  },
  {
    "capacity": 15954.059,
    "output": 4189.23,
    "start": "2023-12-01T12:30:00Z",
  },
]
`;
